src/main/java/subway/line/Line.java

```diff
@@ -11,26 +16,21 @@ public class Line {
     private Long id;
     @Column(length = 20, nullable = false)
     private String name;
-
     @Column(length = 20, nullable = false)
     private String color;
 
-    @OneToOne
-    @JoinColumn(name = "UP_STATION_ID")
-    private Station upStation;
-
-    @OneToOne
-    @JoinColumn(name = "DOWN_STATION_ID")
-    private Station downStation;
-
+    private Long upStationId;
+    private Long downStationId;
+    @OneToMany(mappedBy = "line", fetch = FetchType.LAZY, orphanRemoval = true)
```

Comment:
```suggestion
    @OneToMany(mappedBy = "line", fetch = FetchType.LAZY, cascade = CascadeType.PERSIST, orphanRemoval = true)
```
Line 에서 구간 삭제를 구현해보라고 말씀을 드렸는데, cascade 옵션과 orphanRemoval 옵션을 같이 사용하여 고아객체를 삭제하도록 구현할 수 있습니다. [참고](https://velog.io/@yuseogi0218/JPA-CascadeType.REMOVE-vs-orphanRemoval-true)

물론 이런 옵션은 고민을 많이 하고 추가해야하는 데, 현재 요구사항에서는 `구간은 노선에 종속적`이라 보이기 때문에 이런형태로 구현해도 좋습니다 😄 
해당 옵션을 추가해주면 `SectionRepository.delete(section)` 대신 `line.removeSection(section)` 과 같은 형태로 조금 더 도메인 로직에 집중하도록 구현할 수 있습니다 👍

Reply:
Line안의 로직으로 아래와 같이 옮겼습니다.
```java
    public void removeSection(final Long stationId) {
        final Section deleteSection = this.sections.stream()
                .filter(s -> s.getDownStation().getId().equals(stationId))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("역을 찾을 수 없습니다."));

        this.sections.remove(deleteSection);
    }
```